FireboltNETSDK/Client/FireboltStreamingDataReader.cs

```diff
@@ -0,0 +1,197 @@
+
+        /// <inheritdoc/>
+        public override string GetDataTypeName(int ordinal)
+        {
+            return _metas[ordinal].Type;
+        }
+
+        /// <inheritdoc/>
+        public override IEnumerator GetEnumerator()
+        {
+            return new DbEnumerator(this);
+        }
+
+        /// <inheritdoc/>
+        public override string GetName(int ordinal)
+        {
+            return _metas[ordinal].Name;
+        }
+
+        /// <inheritdoc/>
+        public override int GetOrdinal(string name)
+        {
+            return _metas.FindIndex(m => m.Name == name);
+        }
+
+        /// <inheritdoc/>
+        public override object GetValue(int ordinal)
+        {
+            return GetValueSafely(ordinal) ?? throw new InvalidOperationException($"Column ${ordinal} is null");
+        }
+
+        /// <inheritdoc/>
+        public override int GetValues(object[] values)
+        {
+            List<object?> row = _currentRow ?? new List<object?>();
+            int n = Math.Min(row.Count, values.Length);
+            for (int i = 0; i < n; i++)
+            {
+                values[i] = GetValue(i);
+            }
+            return n;
+        }
+
+        /// <inheritdoc/>
+        public override bool IsDBNull(int ordinal)
+        {
+            ValidateRowIndex();
+            return _currentRow != null && _currentRow[ordinal] == null;
+        }
+
+        /// <inheritdoc/>
+        public override bool Read()
+        {
+            if (IsStreamClosed()) return false;
+            if (_currentRowQueue.Count == 0 && !FetchRow())
+                return false;
+
+            _currentRow = _currentRowQueue.Dequeue();
+            _currentRowIndex++;
+
+            return true;
+        }
+
+        protected override List<object?>? GetRow(int ordinal)
+        {
+            if (ordinal < 0 || ordinal > _currentRow?.Count - 1)
+            {
+                throw new InvalidOperationException($"Column ${ordinal} does not exist");
+            }
+            return _currentRow;
+        }
+
+        protected override ColumnType GetColumnType(int ordinal)
+        {
+            return ColumnType.Of(TypesConverter.GetFullColumnTypeName(_metas[ordinal]));
+        }
+
+        private bool FetchRow()
+        {
+            if (IsStreamClosed()) return false;
+
+            var line = _streamReader.ReadLine();
+            if (string.IsNullOrWhiteSpace(line)) throw new FireboltException("Failed to read line from stream");
+
+            var json = JsonConvert.DeserializeObject<StreamingJsonData>(line);
+            var messageType = json?.MessageType;
+
+            switch (messageType)
+            {
+                case "DATA":
+                    json?.Data.ForEach(_currentRowQueue.Enqueue);
+                    return true;
+
+                case "FINISH_SUCCESSFULLY":
+                    _endOfStream = true;
+                    break;
+
+                default:
```

Comment:
let's update it to `case FINISH_WITH_ERRORS: HandleError(...) default: throw Exception("unexpected message type ...")`